content/guides/digging-deeper/route-model-binding.md

```diff
@@ -0,0 +1,279 @@
+
+:::codegroup
+
+```sh
+// title: Install
+npm i @adonisjs/route-model-binding
+```
+
+```sh
+// title: Configure
+node ace configure @adonisjs/route-model-binding
+
+# UPDATE: tsconfig.json { types += "@adonisjs/route-model-binding/build/adonis-typings" }
+# UPDATE: .adonisrc.json { providers += "@adonisjs/route-model-binding/build/providers/RmbProvider" }
+```
+
+```ts
+// title: Register middleware
+/**
+ * Make sure to add the following global middleware inside
+ * the start/kernel.ts file
+ */
+Server.middleware.register([
+  // ...other middleware
+  () => import('@ioc:Adonis/Addons/RmbMiddleware'),
+])
+```
+ 
+:::
+
+
+:::div{class="features"}
+
+- Works with all Database drivers
+- Customizable lookup logic
+
+&nbsp;
+
+- [View on npm](https://www.npmjs.com/package/@adonisjs/route-model-binding)
+- [View on GitHub](https://github.com/adonisjs/route-model-binding)
+
+:::
+
+## Example
+
+Route model binding is a neat way to remove one-liner Lucid queries from your codebase and use conventions to query the database during HTTP requests.
+
+In the following example, we connect the route params `:post` and `:comments` with the arguments accepted by the `show` method.
+
+- The value of the first param from the URL will be used to query the first typed hinted model on the `show` method (i.e., Post).
+- Similarly, the value of the second param will be used to query the second typed hinted model (i.e., Comment).
+
+:::note
+The params and models are connected using the order they appear and not the name. This is because TypeScript decorators have no way to know the names of the arguments accepted by a method.
+:::
+
+```ts
+// Routes file
+Route.get('posts/:post/comments/:comment', 'PostsController.show')
+```
+
+```ts
+// Controller
+import Post from 'App/Models/Post'
+import Comment from 'App/Models/Comment'
+import { bind } from '@adonisjs/route-model-binding'
+
+export default class PostsController {
+  @bind()
+  public async show({}, post: Post, comment: Comment) {
+    return { post, comment }
+  }
+}
+```
+
+
+:::note
+**Are you a visual learner**? Checkout [these screencasts](https://learn.adonisjs.com/series/route-model-binding/introduction) to learn about Route model binding, its setup and usage.
+:::
+
+
+## Basic usage
+Start with the most basic example and then tune up the complexity level to serve different use cases.
+
+In the following example, we will bind the `Post` model with the first parameter in the `posts/:id` route.
+
+```ts
+Route.get('/posts/:id', 'PostsController.show')
+```
+
+```ts
+import Post from 'App/Models/Post'
+import { bind } from '@adonisjs/route-model-binding'
+
+export default class PostsController {
+  @bind()
+  public async show({}, post: Post) {
+    return { post }
+  }
+}
+```
+
+:::warning
+Make sure you always import your model with an `import` and not an `import type`. Otherwise, the `bind` decorator will not be able to retrieve the class of your model and it will not work.
```

Comment:
```suggestion
Ensure you always import your model with an `import` and not an `import type`. Otherwise, the `bind` decorator will not be able to retrieve the class of your model, and it will not work.
```